packages/nuxt/src/app/composables/router.ts

```diff
@@ -183,6 +184,13 @@ export const navigateTo = (to: RouteLocationRaw | undefined | null, options?: Na
     } else {
       location.href = toPath
     }
+    // Within in a Nuxt route middleware handler
+    if (inMiddleware) {
+      // Abort navigation when app is hydrated
+      if (!nuxtApp.isHydrating) {
+        return false
+      }
+    }
```

Comment:
There is a corner case that's not handled here: when the Nuxt app is loaded on a route that has a redirect middleware. The page gets rendered (and is visible during a very short time) before the location changes. If we also return `false` when `nuxtApp.isHydrating`, it renders a 404 error / page (that's visible during a very short time).

There's something we can do: delay the return statement, like so, for example:
```ts
if (!nuxtApp.isHydrating) {
  return false
}
return new Promise(resolve => setTimeout(resolve, 10))
```
But it looks a bit hacky 🤔

Reply:
I wonder what happens if you return a never-resolving promise 🤔

Reply:
I actually think that's what's happening with `return new Promise(resolve => setTimeout(resolve, 10))` as the location changes before the `resolve` function gets called.

I like the idea of returning a never-resolving promise (`return new Promise(() => {})`) for this case 👍